src/main/java/com/microsoft/azure/datalake/store/HttpTransport.java

```diff
@@ -0,0 +1,375 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ * See License.txt in the project root for license information.
+ */
+
+package com.microsoft.azure.datalake.store;
+
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+import com.microsoft.azure.datalake.store.retrypolicies.NonIdempotentRetryPolicy;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.*;
+import java.util.UUID;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
```

Comment:
Do you need to add exclusions in the pom.xml for this?

Reply:
If I add that, the class will be still in the com.microsoft.azure.deltalake package but the rest of the shaded code will import that from wrong package. Idk how to fix that.

Reply:
i mean excludes like in here
https://github.com/trinodb/trino-hadoop-apache/blob/master/pom.xml#L632-L636

not having the exclude can just hide the problem, shouldn't cause one

Reply:
>  but the rest of the shaded code will import that from wrong package. Idk how to fix that.

@electrum did you have some thoughts on that already?

otherwise i think we can try to put the class into the correct package manually ("shade manually")

Reply:
We already shade `com.microsoft.azure` so this should work like the other classes we replace.

Reply:
@electrum 

We have an artifact that is shaded right now: `hadoop-azure-datalake` and we want to replace one class within this relocation. How it should be done properly?

Reply:
nvm. I think that I got it right. PTAL